editor/filesystem_dock.cpp

```diff
@@ -1563,10 +1563,36 @@ void FileSystemDock::_try_duplicate_item(const FileOrFolder &p_item, const Strin
 }
 
 void FileSystemDock::_update_resource_paths_after_move(const HashMap<String, String> &p_renames, const HashMap<String, ResourceUID::ID> &p_uids) const {
-	// Update the paths in ResourceUID, so that UIDs remain valid.
-	for (const KeyValue<String, ResourceUID::ID> &pair : p_uids) {
-		if (p_renames.has(pair.key)) {
-			ResourceUID::get_singleton()->set_id(pair.value, p_renames[pair.key]);
+	for (const KeyValue<String, String> &pair : p_renames) {
+		// Update UID path.
+		if (p_uids.has(pair.key)) {
+			ResourceUID::get_singleton()->set_id(p_uids[pair.key], pair.value);
```

Comment:
```suggestion
		HashMap<String, ResourceUID::ID>::Iterator I = p_uids.find(pair.key);
		if (I) {
			ResourceUID::get_singleton()->set_id(I->value, pair.value);
```
Avoids double lookup.